sage.py

```diff
@@ -1374,74 +1349,60 @@ def parse(unparsed_data, alert_labels=[], slim=False):
     __sev = set()
     data = []
-    connections = dict()
 
     prev = -1
-    for id, d in enumerate(unparsed_data):
-        #print(d)
-        raw = ''
-        
-        try:
-             raw = json.loads(d['result']['_raw'])
-        except:
-            try:
-                raw = json.loads(d['_raw'])
-            except:
-                raw = d
-                
+    for d in unparsed_data:
+        if 'result' in d and '_raw' in d['result']:
+            raw = json.loads(d['result']['_raw'])
+        elif '_raw' in d:
+            raw = json.loads(d['_raw'])
+        else:
+            raw = d
+
         if raw['event_type'] != 'alert':
             continue
-        #app_proto = raw['app_proto']
-        host = ''
-        
-        try:
+
+        if 'host' in raw:
             host = raw['host']
-        except:
-            try:
-                host = d['host'][3:]
-            except:
-                host = 'dummy'
-                
+        elif 'host' in d:
+            host = d['host'][3:]
+        else:
+            host = 'dummy'
 
-        #print(host)
         ts = raw['timestamp']
-        dt = datetime.datetime.strptime(ts, '%Y-%m-%dT%H:%M:%S.%f%z')# 2018-11-03T23:16:09.148520+0000
-        DIFF = 0.0 if prev== -1 else round((dt - prev).total_seconds(),2)
+        dt = datetime.datetime.strptime(ts, '%Y-%m-%dT%H:%M:%S.%f%z')  # 2018-11-03T23:16:09.148520+0000
+        diff_dt = 0.0 if prev == -1 else round((dt - prev).total_seconds(), 2)
         prev = dt
         
-                
         sig = raw['alert']['signature']
         cat = raw['alert']['category']
-        
         severity = raw['alert']['severity']
 
+        # Filter out the alert that occurs way too often
         if cat == 'Attempted Information Leak' and FILTER:
             continue
+
         srcip = raw['src_ip']
         srcport = None if 'src_port' not in raw.keys() else raw['src_port']
         dstip = raw['dest_ip']
         dstport = None if 'dest_port' not in raw.keys() else raw['dest_port']
 
-        # Filtering out mistaken alerts / uninteresting alerts
+        # Filter out mistaken alerts / uninteresting alerts
         if srcip == badIP or dstip == badIP or cat == 'Not Suspicious Traffic':
             continue
-            
+
         if not slim:
             mcat = get_attack_stage_mapping(sig)
-            data.append((DIFF, srcip, srcport, dstip, dstport, sig, cat, host, dt, mcat))
+            data.append((diff_dt, srcip, srcport, dstip, dstport, sig, cat, host, dt, mcat))
         else:
-            data.append((DIFF, srcip, srcport, dstip, dstport, sig, cat, host, dt))
+            data.append((diff_dt, srcip, srcport, dstip, dstport, sig, cat, host, dt))
 
-        #host_ip.append((host, srcip, dstip))
-       
-        
         __cats.add(cat)
         __ips.add(srcip)
         __ips.add(dstip)
         __hosts.add(host)
         __sev.add(severity)
 
-        
     '''_cats = [(id,c) for (id,c) in enumerate(__cats)]
```

Comment:
This code block is also commented out and I am not sure what to do with that. The sets above (`__cats`, `__ips`, ...) are only used in this place.